GVFS/GVFS.UnitTests/Common/GVFSEnlistmentHealthTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿using GVFS.Common;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class GVFSEnlistmentHealthTests
+    {
+        private readonly char sep = GVFSPlatform.GVFSPlatformConstants.PathSeparator;
+
+        private List<string> fauxGitFiles = new List<string>();
+        private List<string> fauxGitFolders = new List<string>();
+        private List<string> fauxPlaceholderFiles = new List<string>();
+        private List<string> fauxPlaceholderFolders = new List<string>();
+        private List<string> fauxModifiedPathsFiles = new List<string>();
+        private List<string> fauxModifiedPathsFolders = new List<string>();
+        private List<string> fauxSkipWorktreeFiles = new List<string>();
+        private GVFSEnlistmentHealthCalculator enlistmentHealthCalculator;
+        private GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData;
+
+        [TestCase]
+        public void OnlyOneHydratedDirectory()

```

Comment:
This name doesn't really match the test.  I like to name the tests a little more specific so that if it fails it is easier to know why something like `SomeHydratedFilesInSingleDirectoryShouldHaveOtherDirectoriesEqualToZeroPercent`

Reply:
Tracking this in a larger GVFSEnlistmentHealhTests refactoring issue: https://github.com/microsoft/VFSForGit/issues/1299